packages/web/src/js/Password.ts

```diff
@@ -0,0 +1,59 @@
+import SelectorEngine from './dom/SelectorEngine';
+import EventHandler from './dom/EventHandler';
+import BaseComponent from './BaseComponent';
+import { getElement } from './utils/index';
+
+const PASSWORD_TOGGLE_SELECTOR = '[data-toggle="password"]';
+const PASSWORD_ARIA_PRESSED = 'aria-pressed';
+const PASSWORD_ARIA_LABEL = 'aria-label';
+const PASSWORD_FIELD_ELEMENT = 'input';
+
+class Password extends BaseComponent {
+  isShown: boolean;
+
+  constructor(element: HTMLElement) {
+    super(element);
+
+    this.isShown = false;
+  }
+
+  show(target: HTMLElement) {
+    target.setAttribute(PASSWORD_ARIA_PRESSED, 'true');
+    target.setAttribute(PASSWORD_ARIA_LABEL, 'Hide password');
+    SelectorEngine.findOne(PASSWORD_FIELD_ELEMENT, target.parentElement as HTMLElement).setAttribute('type', 'text');
+
+    this.isShown = true;
+  }
+
+  hide(target: HTMLElement) {
+    target.setAttribute(PASSWORD_ARIA_PRESSED, 'false');
+    target.setAttribute(PASSWORD_ARIA_LABEL, 'Show password');
+    SelectorEngine.findOne(PASSWORD_FIELD_ELEMENT, target.parentElement as HTMLElement).setAttribute('type', 'password');
+
+    this.isShown = false;
+  }
+
+  toggle(relatedTarget: HTMLElement) {
+    this.isShown ? this.hide(relatedTarget) : this.show(relatedTarget);
+  }
+}
+
+function handlePasswordClick() {
+  const target = getElement(this);
+
+  if (target) {
+    const password = new Password(target);
+    password.toggle(target);
+  }
+}
+
+// When document content is loaded
```

Comment:
Do you think the comments are necessary…? 🤔 I believe the code itself is readable enough (+1 for that). The same goes for `Header` handler.

Reply:
Ok, this was fine during step-by-step implementation for me to better grab that logic. Now is readable enough. I agree. Removing them.